api/v1beta2/awsmachine_types.go

```diff
@@ -64,6 +64,8 @@ const (
 )
 
 // AWSMachineSpec defines the desired state of an Amazon EC2 instance.
+// +kubebuilder:validation:XValidation:rule="!has(self.capacityReservationId) || !has(self.marketType) || self.marketType != 'Spot'",message="capacityReservationId may not be set when marketType is Spot"
+// +kubebuilder:validation:XValidation:rule="!has(self.capacityReservationId) || !has(self.spotMarketOptions)",message="capacityReservationId cannot be set when spotMarketOptions is specified"
```

Comment:
I'll need to check the docs (but maybe @JoelSpeed knows), if we are tightening the validation rules is this considered a breaking behavioural change? Guess it depends on if it breaks currently.

@athiruma -  whats the current behaviour if i set `capacityReservationId` and specify spot? Does it currently error in this situation?

Reply:
Yes, it will raise an error.

Reply:
If the capacity reservation ID field has already shipped within a release, then yes, this could cause an issue for users who have already started using the field

Given that EC2 will reject creating machines with this configuration already, we are bringing the validation earlier here realistically, and ratcheting should also help with invalid combinations if folks are updating the objects.

I'm inclined to say this is likely to be fairly safe depending on how long ago this shipped

Reply:
It has shipped in CAPA v2.8.1 a couple of weeks ago.
I think if we backport it to v2.8.z and then release it in v2.9.0 we should be fine right?

Reply:
Sounds like we are all good then.

Reply:
Yeah considering this would already fail anyway at a later stage.
Happy to approve then @richardcase ?